exporter/sumologicexporter/exporter.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicexporter
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+type sumologicexporter struct {
+	config *Config
+}
+
+func initExporter(cfg *Config) (*sumologicexporter, error) {
+	switch cfg.LogFormat {
+	case JSONFormat:
+	case TextFormat:
+	default:
+		return nil, fmt.Errorf("unexpected log format: %s", cfg.LogFormat)
+	}
+
+	switch cfg.MetricFormat {
+	case GraphiteFormat:
+	case Carbon2Format:
+	case PrometheusFormat:
+	default:
+		return nil, fmt.Errorf("unexpected metric format: %s", cfg.MetricFormat)
+	}
+
+	switch cfg.CompressEncoding {
+	case GZIPCompression:
+	case DeflateCompression:
+	case NoCompression:
+	default:
+		return nil, fmt.Errorf("unexpected compression encoding: %s", cfg.CompressEncoding)
+	}
+
+	if len(cfg.HTTPClientSettings.Endpoint) == 0 {
+		return nil, fmt.Errorf("endpoint is not set")
```

Comment:
`Errorf` seems wrong here, you don't have any parameters to format. Use `errors.New` instead (perhaps even as a var, so that tests can use it)